clients/src/test/java/org/apache/kafka/clients/admin/MockAdminClient.java

```diff
@@ -560,10 +529,15 @@ synchronized private Config getResourceDescription(ConfigResource resource) {
             }
             case TOPIC: {
                 TopicMetadata topicMetadata = allTopics.get(resource.name());
-                if (topicMetadata == null) {
-                    throw new UnknownTopicOrPartitionException();
+                if (topicMetadata != null && !topicMetadata.markedForDeletion) {
+                    if (topicMetadata.fetchesRemainingUntilVisible > 0)
+                        topicMetadata.fetchesRemainingUntilVisible = Math.max(0, topicMetadata.fetchesRemainingUntilVisible - 1);
+                    else return new Config(topicMetadata.configs.entrySet().stream()
+                                .map(entry -> new ConfigEntry(entry.getKey(), entry.getValue()))
+                                .collect(Collectors.toList()));
```

Comment:
nit: I think that we can use `toConfigObject(topicMetadata.configs)` here.

Reply:
It has been merged when I notice your comment. Let me address it in another PR :)

Reply:
#8889